src/Uno.Toolkit.RuntimeTests/Tests/LeakTest.cs

```diff
@@ -0,0 +1,251 @@
+﻿#if !WINDOWS_UWP && !NET5_0_WINDOWS10_0_18362
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+//using Private.Infrastructure;
+//using Uno.Extensions;
+using Uno.UI;
+using Uno.UI.RuntimeTests;
+//using Uno.UI.RuntimeTests.Tests.Windows_UI_Xaml.Controls;
+using Windows.UI.Core;
+
+#if __MACOS__
+using AppKit;
+#elif __IOS__
+using UIKit;
+#endif
+
+#if IS_WINUI
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using Microsoft.UI.Xaml.Controls.Primitives;
+#else
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+#endif
+
+namespace Uno.Toolkit.RuntimeTests.Tests
+{
+	[TestClass]
+	[Ignore("#429: Disabled as the tests are failing even for the basic types. ")]
+	public class LeakTest
+	{
+		private static DependencyProperty VisualStateManagerVisualStateManagerProperty { get; } = (DependencyProperty)typeof(VisualStateManager)
+			.GetProperty("VisualStateManagerProperty", BindingFlags.NonPublic | BindingFlags.Static)
+			.GetValue(null);
+
+		[TestMethod]
+		[RunsOnUIThread]
+#if __MACOS__
+		[Ignore("Currently fails on macOS, part of #9282! epic")]
+#endif
+		[DataRow(typeof(TextBox), 15)]
+		[DataRow(typeof(Button), 15)]
+		public async Task When_Add_Remove(object controlTypeRaw, int count)
+		{
+//#if TRACK_REFS
+//			var initialInactiveStats = Uno.UI.DataBinding.BinderReferenceHolder.GetInactiveViewReferencesStats();
+//			var initialActiveStats = Uno.UI.DataBinding.BinderReferenceHolder.GetReferenceStats();
+//#endif
+
+			Type GetType(string s)
+				=> AppDomain.CurrentDomain.GetAssemblies().Select(a => a.GetType(s)).Where(t => t != null).First()!;
+
+			var controlType = controlTypeRaw switch
+			{
+				Type ct => ct,
+				string s => GetType(s),
+				_ => throw new InvalidOperationException()
+			};
+
+			var _holders = new ConditionalWeakTable<DependencyObject, Holder>();
+			void TrackDependencyObject(DependencyObject target) => _holders.Add(target, new Holder(HolderUpdate));
+
+			var maxCounter = 0;
+			var activeControls = 0;
+			var maxActiveControls = 0;
+
+			var rootContainer = new ContentControl();
```

Comment:
Just comparing this to https://github.com/unoplatform/uno/blob/master/src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml/Given_FrameworkElement_And_Leak.cs

They have a `_holder.Reset()` before setting the `rootContainer`, could it maybe a reason they're all failing?

Reply:
that was added recently; but doesnt help since even the first test is already failing...